signer/local_cosigner.go

```diff
@@ -146,15 +150,67 @@ func (cosigner *LocalCosigner) GetID() int {
 	return cosigner.key.ID
 }
 
-// GetAddress returns the RPC URL of the cosigner
+// GetAddress returns the GRPC URL of the cosigner
 // Implements Cosigner interface
 func (cosigner *LocalCosigner) GetAddress() string {
 	return cosigner.address
 }
 
-// Sign the sign request using the cosigner's share
-// Return the signed bytes or an error
+// GetEphemeralSecretParts
+// Implements Cosigner interface
+func (cosigner *LocalCosigner) GetEphemeralSecretParts(
+	hrst HRSTKey) (*CosignerEphemeralSecretPartsResponse, error) {
+	res := &CosignerEphemeralSecretPartsResponse{
+		EncryptedSecrets: make([]CosignerEphemeralSecretPart, 0, len(cosigner.peers)-1),
+	}
+	for _, peer := range cosigner.peers {
+		if peer.ID == cosigner.GetID() {
+			continue
+		}
+		secretPart, err := cosigner.getEphemeralSecretPart(CosignerGetEphemeralSecretPartRequest{
+			ID:        peer.ID,
+			Height:    hrst.Height,
+			Round:     hrst.Round,
+			Step:      hrst.Step,
+			Timestamp: time.Unix(0, hrst.Timestamp),
+		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		res.EncryptedSecrets = append(res.EncryptedSecrets, secretPart)
+	}
+	return res, nil
+}
+
+// SetEphemeralSecretPartsAndSign
 // Implements Cosigner interface
+func (cosigner *LocalCosigner) SetEphemeralSecretPartsAndSign(
+	req CosignerSetEphemeralSecretPartsAndSignRequest) (*CosignerSignResponse, error) {
+	for _, secretPart := range req.EncryptedSecrets {
+		err := cosigner.setEphemeralSecretPart(CosignerSetEphemeralSecretPartRequest{
+			SourceID:                       secretPart.SourceID,
+			SourceEphemeralSecretPublicKey: secretPart.SourceEphemeralSecretPublicKey,
+			EncryptedSharePart:             secretPart.EncryptedSharePart,
+			SourceSig:                      secretPart.SourceSig,
+			Height:                         req.HRST.Height,
+			Round:                          req.HRST.Round,
+			Step:                           req.HRST.Step,
+			Timestamp:                      time.Unix(0, req.HRST.Timestamp),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	res, err := cosigner.sign(CosignerSignRequest{req.SignBytes})
+	return &res, err
+}
+
+// sign the sign request using the cosigner's share
+// Return the signed bytes or an error
+// Implements Cosigner interface # Comment is this really true? Doesnt it implement ThresholdEd25519Signature
 func (cosigner *LocalCosigner) sign(req CosignerSignRequest) (CosignerSignResponse, error) {
```

Comment:
These should move out to a new file for the "local" threshold ed25519 signature implementation.